samcli/commands/init/init_templates.py

```diff
@@ -171,3 +173,46 @@ def is_dynamic_schemas_template(self, package_type, app_template, runtime, base_
             if option.get("appTemplate") == app_template:
                 return option.get("isDynamicTemplate", False)
         return False
+
+    def get_preprocessed_manifest(self):
+        """
+        This method get the manifest cloned from the git repo and preprocessed it.
+
+        Raises:
+            InvalidInitTemplateError: Exception raise when template is invalid or missing a parameter
+
+        Returns:
+            dict: This is preprocessed manifest with the use_case as key
+        """
+        manifest_dump = requests.get(self._manifest_url)
```

Comment:
we could for check for response object instead 

```
resp = requests.get(self._manifest_url)

if not resp.ok:
    ....
```

Reply:
Also please note that we just started reviewing the tech doc and some iteration might be expected for this pr